core/trino-spi/src/main/java/io/trino/spi/type/TypeUtils.java

```diff
@@ -69,30 +69,33 @@ public static void writeNativeValue(Type type, BlockBuilder blockBuilder, @Nulla
         if (value == null) {
             blockBuilder.appendNull();
         }
-        else if (type.getJavaType() == boolean.class) {
-            type.writeBoolean(blockBuilder, (Boolean) value);
-        }
-        else if (type.getJavaType() == double.class) {
-            type.writeDouble(blockBuilder, ((double) value));
-        }
-        else if (type.getJavaType() == long.class) {
-            type.writeLong(blockBuilder, ((long) value));
-        }
-        else if (type.getJavaType() == Slice.class) {
-            Slice slice;
-            if (value instanceof byte[]) {
-                slice = Slices.wrappedBuffer((byte[]) value);
+        else {
```

Comment:
you can unwrap this `else` by adding a return statement in the `if` branch

Reply:
updated